### PR TITLE
feat: organize settings dialog with tabs

### PR DESCRIPTION
**Pull request by cristianoventura**
## Description

This PR changes the Settings dialog by separating sections into tabs. This provides a cleaner user interface as we add more functionalities to the Studio and enable users with custom settings.

## How to Test

- Make the state of the settings dialog invalid
- Attempt to submit
- Check that tabs with errors are automatically selected and, inputs with errors automatically get focus

## Checklist

- [x] I have performed a self-review of my code.
- [ ] I have added tests for my changes.
- [x] I have run linter locally (`npm run lint`) and all checks pass.
- [x] I have run tests locally (`npm test`) and all tests pass.
- [x] I have commented on my code, particularly in hard-to-understand areas.

## Screenshots (if appropriate):

https://github.com/user-attachments/assets/a01c999c-a401-4a8e-a255-2ec6c0097c92

### CODE REVIEW

File: src/components/Settings/SettingsDialog.tsx [hunk 1]
```diff
@@ -1,23 +1,36 @@
...
+import { ExclamationTriangleIcon } from '@radix-ui/react-icons'
+import { ButtonWithTooltip } from '../ButtonWithTooltip'
+import { findIndex, sortBy } from 'lodash-es'
 
 type SettingsDialogProps = {
   open: boolean
   onOpenChange: (open: boolean) => void
 }
 
+const tabs = [
```
Comment (cristianoventura) [inline src/components/Settings/SettingsDialog.tsx:20]: I extracted the content of the tabs to an array, so we can loop through and select which one to focus on after an error happens, following the order they are rendered in the UI. This ensures that the user is taken to the very first error at the top of the form. 

This loop happens in the `onInvalid` method on L72.

Comment (going-confetti) [reply src/components/Settings/SettingsDialog.tsx:20]: > so we can loop through and select which one to focus on after an error happens

Looks like saving on blur/change can help us avoid the problem altogether: we won't need to show the warning icon and focus on a broken tab, and the user won't need to remember to save when the changes they made are on a different tab

**Review by Llandy3d — DISMISSED** [review]
Looks good!

I wonder if it makes sense for the `Recorder` section to show greyed out the inputs for selecting a specific browser... it feels like the information that you can actually do that it's hidden away 🤔

**Review by going-confetti — APPROVED** [review]
LGTM!
@cristianoventura and I talked about auto-saving changes and it looks like it'll be a separate improvement later on, because of how settings are designed (for example, changing the proxy mode changes how the form should be validated)

**Pull request merged**